File: client/lua/krpc/test/test_platform.lua
local luaunit = require 'luaunit'
local class = require 'pl.class'
local platform = require 'krpc.platform'

local TestPlatform = class()

function TestPlatform:test_bytelength()
  luaunit.assertEquals(0, (''):len())
  luaunit.assertEquals(3, ('foo'):len())
  luaunit.assertEquals(3, ('\226\132\162'):len())
  --luaunit.assertEquals(3, ('\u2122'):len())
end

function TestPlatform:test_hexlify()
  luaunit.assertEquals(platform.hexlify(''), '')
  luaunit.assertEquals(platform.hexlify('\0\1\2'), '000102')
  luaunit.assertEquals(platform.hexlify('\255'), 'ff')
end

function TestPlatform:test_unhexlify()
  luaunit.assertEquals(platform.unhexlify(''), '')
  luaunit.assertEquals(platform.unhexlify('000102'), '\0\1\2')
  luaunit.assertEquals(platform.unhexlify('ff'), '\255')
end

return TestPlatform

File: client/python/krpc/test/test_platform.py
import unittest
from krpc.platform import bytelength, hexlify, unhexlify


class TestPlatform(unittest.TestCase):
    def test_bytelength(self):
        self.assertEqual(0, bytelength(''))
        self.assertEqual(3, bytelength('foo'))
        self.assertEqual(3, bytelength(b'\xe2\x84\xa2'.decode('utf-8')))
        self.assertEqual(3, bytelength(u'\u2122'))

    def test_hexlify(self):
        self.assertEqual(hexlify(b''), '')
        self.assertEqual(hexlify(b'\x00\x01\x02'), '000102')
        self.assertEqual(hexlify(b'\xFF'), 'ff')

    def test_unhexlify(self):
        self.assertEqual(unhexlify(''), b'')
        self.assertEqual(unhexlify('000102'), b'\x00\x01\x02')
        self.assertEqual(unhexlify('ff'), b'\xFF')

if __name__ == '__main__':
    unittest.main()
